### PR TITLE
[Network] az network firewall policy rule-collection-group rule-collection: support multiple DNAT rules and ip groups settings

### PR DESCRIPTION
**Pull request by MyronFanQiu**
---

Fix Azure/azure-cli#13811
Fix Azure/azure-cli#13631

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S172

File: src/azure-firewall/azext_firewall/custom.py [hunk 1]
```diff
@@ -684,39 +691,60 @@ def add_azure_firewall_policy_filter_rule(cmd, resource_group_name, firewall_pol
     if target_rule_collection is None:
         raise CLIError("Cannot find corresponding rule.")
 
+    if target_rule_collection.rule_collection_type == "FirewallPolicyFilterRule" and rule_type == 'NatRule':
+        raise CLIError("FirewallPolicyFilterRule doesn't support Nat rule.")
+
+    if target_rule_collection.rule_collection_type == "FirewallPolicyNatRule" and rule_type in ['NetworkRule',
+                                                                                                'ApplicationRule']:
+        raise CLIError("FirewallPolicyNatRule doesn't support Network rule and Application rule.")
```
Comment (jsntcy) [inline src/azure-firewall/azext_firewall/custom.py:699]: supports neither ... nor ...

File: src/azure-firewall/azext_firewall/_params.py [hunk 2]
```diff
@@ -169,18 +170,31 @@ def load_arguments(self, _):
...
+    with self.argument_context('network firewall policy rule-collection-group collection', arg_group='Common Rule') as c:
+        c.argument('description', help='The description of rule.')
+        c.argument('destination_addresses', nargs='+', help="Space-separated list of destination IP addresses.")
+        c.argument('source_addresses', nargs='+', help="Space-separated list of source IP addresses.")
+        c.argument('rule_name', options_list=['--rule-name'], help='The name of rule')
+        c.argument('rule_type', options_list=['--rule-type'], arg_type=get_enum_type(["ApplicationRule", "NetworkRule", "NatRule"]), help='The type of rule')
+        c.argument('destination_ports', nargs='+', help="Space-separated list of destination ports. This argument is supported for Nat and Network Rule.")
+        c.argument('ip_protocols', nargs='+', arg_type=get_enum_type(["TCP", "UDP", "Any", "ICMP"]),
+                   help="Space-separated list of IP protocols. This argument is supported for Nat and Network Rule.")
+        c.argument('source_ip_groups', nargs='+', arg_group='Common Rule', validator=validate_ip_groups,
```
Comment (jsntcy) [inline src/azure-firewall/azext_firewall/_params.py:183]: arg_group is redundancy

File: src/azure-firewall/azext_firewall/custom.py (hunk 1 above)
Comment (MyronFanQiu) [reply src/azure-firewall/azext_firewall/custom.py]: fix

File: src/azure-firewall/azext_firewall/_params.py [hunk 3]
```diff
@@ -169,18 +170,31 @@ def load_arguments(self, _):
...
+                   help='Space-separated list of name or resource id of source IpGroups.')
+
+    with self.argument_context('network firewall policy rule-collection-group collection', arg_group='Nat Rule') as c:
+        c.argument('translated_address', arg_group='Nat Rule', help='Translated address for this NAT rule collection.')
```
Comment (jsntcy) [inline src/azure-firewall/azext_firewall/_params.py]: arg_group is redundancy

File: src/azure-firewall/azext_firewall/_params.py (hunk 2 above)
Comment (MyronFanQiu) [reply src/azure-firewall/azext_firewall/_params.py]: fix

File: src/azure-firewall/azext_firewall/_params.py (hunk 3 above)
Comment (MyronFanQiu) [reply src/azure-firewall/azext_firewall/_params.py]: fix

**Pull request merged**